strategy/example2.go

```diff
@@ -0,0 +1,147 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+)
+
+// =========================== ANALYZER =================================
```

Comment:
Let's maybe follow a common commentary style guide, https://golang.org/doc/effective_go.html#commentary ?

Reply:
This is not package/ function comment. It was just meant to separate the components for visibility. 
Do you suggest to remove it ?

Reply:
Maybe having this with additional line comments over your interface or methods will be helpful.

Reply:
ok.